targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/serial_api.c

```
@@ -0,0 +1,452 @@
+        stdio_uart_inited = 1;
+        memcpy(&stdio_uart, obj, sizeof(serial_t));
+    }
+}
+
+void serial_free(serial_t *obj)
+{
+      uart_data[obj->index].serial_irq_id = 0;
+}
+
+void serial_baud(serial_t *obj, int baudrate)
+{
+    /*
+     * The MPS2 has a simple divider to control the baud rate.
+     * The formula is:
+     * Baudrate = PCLK / BAUDDIV  where  PCLK = SystemCoreClock and
+     *                                   BAUDDIV is the desire baudrate
+     *
+     * So, if the desired baud rate is 9600 the calculation will be:
+     * Baudrate = SystemCoreClock / 9600;
+     */
+
+    /* Check to see if minimum baud value entered */
+    int baudrate_div = 0;
+
+    if (baudrate == 0) {
+        error("Invalid baudrate value");
+        return;
+    }
+
+    baudrate_div = SystemCoreClock / baudrate;
+
+    if (baudrate >= 16) {
+        switch ((int)obj->uart) {
+            case UART_0:
+                CMSDK_UART0->BAUDDIV = baudrate_div;
+                break;
+            case UART_1:
+                CMSDK_UART1->BAUDDIV = baudrate_div;
+                break;
+            case UART_2:
+                CMSDK_UART2->BAUDDIV = baudrate_div;
+                break;
+            case UART_3:
+                CMSDK_UART3->BAUDDIV = baudrate_div;
+                break;
+            case UART_4:
+                CMSDK_UART4->BAUDDIV = baudrate_div;
+                break;
+            default:
+                error("Invalid uart object");
+                break;
+        }
+    } else {
+        error("Invalid baudrate value");
+    }
+
+}
+
+void serial_format(serial_t *obj, int data_bits,
+                   SerialParity parity, int stop_bits)
+{
+    /*
+     *  The CMSDK APB UART is a simple design that supports 8-bit communication
+     *  without parity, and is fixed at one stop bit per configuration.
+     *  Ref. DDI0479C_cortex_m_system_design_kit_r1p0_trm.pdf
+     */
+    error("serial format function not supported");
```

+1 for failing loudly instead of silently 😄